### PR TITLE
[TEST] Changed test.c to use thread, not process. Remove temp file

### PR DESCRIPTION
**Pull request by benliao1**
As per title of pull request. Removes the need of a temp file in test framework.

### CODE REVIEW

File: tests/test.c [hunk 1]
```diff
@@ -1,61 +1,53 @@
...
+ * It constantly reads from the read end of the pipe, and then prints it to stderr
+ * and stores in test_output. It is canceled by the test in the function end_test().
+ * Arguments:
+ *    args: always NULL
+ * Returns: NULL
+ */
+static void *output_handler(void *args) {
+    size_t curr_size = 1024;
+    size_t num_total_bytes_read = 0;
+    char *curr_ptr = test_output = malloc(curr_size);
```
Comment (AVSurfer123) [inline tests/test.c:24]: bad style. we should never do multiple assignments on same line

File: tests/test.c [hunk 2]
```diff
@@ -1,61 +1,53 @@
...
+/**
+ * This thread prints output to terminal and also copies it to standard output
+ * It constantly reads from the read end of the pipe, and then prints it to stderr
+ * and stores in test_output. It is canceled by the test in the function end_test().
+ * Arguments:
+ *    args: always NULL
+ * Returns: NULL
+ */
+static void *output_handler(void *args) {
+    size_t curr_size = 1024;
```
Comment (AVSurfer123) [inline tests/test.c:22]: should be MAX_LOG_LEN

File: tests/test.c [hunk 3]
```diff
@@ -83,98 +75,66 @@ static void fprintf_delimiter(FILE *stream, char *format, ...) {
...
-
-        // kill the child on Ctrl-C (SIGINT)
-        signal(SIGINT, kill_child);
-
-        // register the clean up handler
-        atexit(cleanup_handler);
+    // save the standard out attached to the terminal for later
+    save_std_out = dup(fileno(stdout));
+    
+    // route stdin to read end of pipe
```
Comment (AVSurfer123) [inline tests/test.c:104]: comment should be other way around. read end gets routed to stdin

File: tests/test.c [hunk 4]
```diff
@@ -83,98 +75,66 @@ static void fprintf_delimiter(FILE *stream, char *format, ...) {
...
+    if (dup2(pipe_fd[0], fileno(stdin)) == -1) {
+        fprintf(stderr, "dup2 stdin to read end of pipe: %s\n", strerror(errno));
+    }
+    // route stdout to write end of pipe
```
Comment (AVSurfer123) [inline tests/test.c:108]: comment should be other way around

File: tests/test.c (hunk 1 above)
Comment (benliao1) [reply tests/test.c:24]: 👌

File: tests/test.c (hunk 2 above)
Comment (benliao1) [reply tests/test.c:22]: 👌

File: tests/test.c (hunk 3 above)
Comment (benliao1) [reply tests/test.c:104]: 👌

File: tests/test.c (hunk 4 above)
Comment (benliao1) [reply tests/test.c:108]: 👌

**Pull request merged**